specs-derive/src/impl_saveload.rs

```diff
@@ -0,0 +1,496 @@
+//! Contains implementations for #[derive(Saveload)]
+//! which derives both `IntoSerialize` and `FromDeserialize`.
+//!
+//! Since this requires a proxy "Data" `struct`/`enum`, these two need
+//! to be derived together or it begins requiring unwieldy attribute explosion.
+//!
+//! Currently there is one major limitation to this implementation: it cannot handle
+//! generic types such as `struct Foo<T> { x: T, e: Entity }`. I'm not good enough at
+//! using `syn` to iron this out, but since specs requires explicit references to non-generic
+//! components in system data, it can generally be worked around by newtyping like
+//! `struct TypedFoo(Foo<u32>);` and the like.
```

Comment:
This is how I handled generics on the derive target: https://github.com/WaDelma/bob/blob/master/bob/src/lib.rs#L400
The helper function `add_ty_params` is at: https://github.com/WaDelma/bob/blob/master/bob/src/lib.rs#L844

The code is bit messy and I wouldn't write the helper like that again, but it should give some idea how to do it.